### PR TITLE
Use generic type to fix stuff

### PR DESCRIPTION
**Pull request by Eskaan**
My attempt to fix your code using a generic type. I already fiddled with this generic deserializing stuff and the trait you might be searching for is `DeserializeOwned`. You may or may not merge this (**I did not test it**)(you might want to find your own solution), but `make test-full` checks out for me.

### CODE REVIEW

**Comment by Eskaan** [comment]
Tested the code and there were some logic issues left. Fixed those too and I suppose everything should work now.

File: leftwm/src/utils/file_handler.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ use ron::{
     ser::{to_string_pretty, PrettyConfig},
     Options,
 };
-use serde::Deserialize;
+use serde::de::DeserializeOwned;
```
Comment (VuiMuich) [inline leftwm/src/utils/file_handler.rs:8]: Ah nice!
This and ...

File: leftwm/src/utils/file_handler.rs [hunk 2]
```diff
@@ -83,21 +81,21 @@ pub fn load_config_file(_config_filename: &Option<PathBuf>) -> Result<Config> {
...
-        read_ron_config(&path)
+        read_ron_config(path)
     } else {
         let from_file = toml::from_str(&fs::read_to_string(path)?)?;
         Ok(from_file)
     }
 }
 
-fn read_ron_config(
+fn read_ron_config<T: DeserializeOwned>(
```
Comment (VuiMuich) [inline leftwm/src/utils/file_handler.rs:94]: ... this were exactly the missing links for me!

**Review by VuiMuich — APPROVED** [review]

**Comment by VuiMuich** [comment]
Thanks a ton. I had a vague idea how it would need to work, but as said in the other PR, my Rust got really rusty :stuck_out_tongue:

**Pull request merged**